internal/rest/resources/control.go

```diff
@@ -210,20 +211,23 @@ func joinWithToken(state state.State, r *http.Request, req *internalTypes.Contro
 	}
 
 	// Get a client to the target address.
-	var lastErr error
+	var allErrors []error
 	var joinInfo *internalTypes.TokenResponse
 	for _, addr := range token.JoinAddresses {
 		url := api.NewURL().Scheme("https").Host(addr.String())
 
 		cert, err := shared.GetRemoteCertificate(url.String(), "")
 		if err != nil {
 			logger.Warn("Failed to get certificate of cluster member", logger.Ctx{"address": url.String(), "error": err})
+			allErrors = append(allErrors, err)
 			continue
 		}
 
 		fingerprint := shared.CertFingerprint(cert)
 		if fingerprint != token.Fingerprint {
 			logger.Warn("Cluster certificate token does not match that of cluster member", logger.Ctx{"address": url.String(), "fingerprint": fingerprint, "expected": token.Fingerprint})
```

Comment:
@masnax can we consider this a break in the protocol if one of the remote cluster members returns a different certificate? Because then I would argue we can safely do `return nil, <err>` as something is off with the remote member.

Reply:
Same as if `internalClient.New` below returns an error.

Reply:
Indeed, I can't think of a situation where the cluster certificate doesn't match on one member, but might on another.

Reply:
@claudiubelu If you could please replace this block with an `return nil, fmt.Error...` then we should be able to backport this into `v2` and get a new release out tomorrow.